### ISSUE TITLE
Apparent failed install of controller and panel

### ISSUE THREAD

**Issue by mzuniga51** [tagyoureit/nodejs-poolController#353]
Hello folks.  I am a noob in all things Raspberry Pi but gave it a try to install the Pool Controller following the very detailed instructions MyAZPool provided in TFP.   Installation of both applications got hung after some errors and when launching the web interface I find several strange behaviors, maybe consistent with a failed installation.  For instance, my pool appear off when it is actually on, and the pump is actually running.  I can control light, circuits, pumps and I can see chemistry.

I've tried installation a few times, one fixing npm audit errors (I realize I should not do that now) but the errors persists in either cases.

Per tagyoureit instruction I'm attaching the packet capture plus the applications installation logs.  

Any help will be much appreciated!  

[2021-10-10T01_51_47_875Z-debug.log](https://github.com/tagyoureit/nodejs-poolController/files/7318068/2021-10-10T01_51_47_875Z-debug.log)
[2021-10-10T01_58_16_401Z-debug.log](https://github.com/tagyoureit/nodejs-poolController/files/7318069/2021-10-10T01_58_16_401Z-debug.log)
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/7318070/replay.zip)

<img width="817" alt="136494302-d3c285b0-c35c-4ac1-a3a3-b1a7b3db2dff" src="https://user-images.githubusercontent.com/91519307/136711322-222c8dab-fc60-4204-9f23-08e6fa44f071.png">

<img width="812" alt="136494320-716dcd14-b31b-408b-9301-7e8e973a6ae9" src="https://user-images.githubusercontent.com/91519307/136711326-394136c0-34b1-4f34-98a9-8798870876b9.png">

**Pool Equipment**
 - Intellicenter i10D (dual pool and spa equipment)
 - Controller Firmware Version 1.047
 - Pump(s) manufacturer and model: 2x IntelliFlow 2 VST 011056
 - Chlorinator: IntelliChlor-40 (pool) IntelliChlor-20 (spa)
 - Heater(s): 2x Master
 - Chemical controller: 2x Intellichem
 - 2x CO2 feeders
 - Jandy Valves with two actuators (pool bubbler and spa jets)

**Raspberry Pi**
Raspberry Pi 3 B+
Raspberry Pi OS 5.10
USB to RS485 adapter https://www.amazon.com/dp/B081MB6PN2?psc=1&ref=ppx_yo2_dt_b_product_details
connected with Cat5e shielded cable to Intellicenter Com port

**Desktop:**
 - OS: macOS Big Sur 11.6
 - Browser: Safari
 - Version 15

**Smartphone:**
 - Device: Iphone 11 Max
 - OS: iOS 15.0.1
 - Browser: Safari

**Comment by rstrouse** [comment]
Ok so you got me a bit excited with the heaters.  So you do have these configured incorrectly.  You need to delete these in IntelliCenter and add them as Gas Heaters.   If you do not IntelliCenter will be doing work behind the scenes to wake them up.  

While you may not see this on the surface, your IntelliCenter is frantically sending messages to the heater.  The communication setup for RS485 heaters outside of UltraTemp are not widely shipping yet.  In fact the boards don't even have the port or the transceiver chip.  So when you are connected to the Gas Heater port on the board (which you are) then you must choose the Gas Heater.  Somebody at Pentair should be slapped for not making that clear.  The strange thing is that you gain nearly nothing by being on the RS485 bus other than creating more traffic.

With the metric settings I will figure out where why the spa temp is coming in in F.  I am guessing your spa isn't 100C.

The No Communication error on the IntelliChem is definitely a bug.  It is just not clearing the message when the IntelliChem finally responds.  During startup it is common for IntelliChem (and IntelliChlor) to not respond right away.

For the body circuit I will put together some instructions for a very specific replay capture that I would like you to do.  We have only seen one other IntelliCenter i10D and it was set up differently than yours.   So we are probably missing a bit on the status message that tells us whether the pool circuit is engaged.

**Comment by rstrouse** [comment]
Your heaters are not correctly configured.  Even though they are MasterTemp heaters they are not correctly defined.  Only MasterTemp heaters that have RS485 connections should be defined the way you have them.  The Max-e-Therm id is simply because njsPC has not seen a response from the heater yet.  If it were to see a response it would change this to the proper model as Max-e-therm and MasterTemp share the same boards.  It should however start with the Pentair version instead of the Sta-Rite version.  Or simply not set it until it actually sees a response.

When you add them back you need to add them as Gas not MasterTemp.  If you don't IntelliCenter will incorrectly attempt to contact them on the RS485 bus.  This results in repeated traffic on the bus.  Every time a message to the heater fails it will send 5 more in rapid succession.  Then after about 1.5 seconds it will start over again to do the same thing.  I don't know why they didn't identify these as MasterTemp RS485 or Max-e-therm RS485 but I assure you the way you have them set up is not desirable.

I will try to get to a fix for the comms error on the IntelliChem and put some instructions together for you to create the replay I need.  It appears that the bit for the pool circuit is not set on the status message for the i10D but it is set on the configuration message.  Wierd.

I also just noticed another thing that is not explainable.  Here is the screenshot you sent and it appears to have a jacked up water temp as well for the spa.
![image](https://user-images.githubusercontent.com/47839015/136854966-705bcf69-f299-4058-88a6-b899f924b61c.png)

**Comment by rstrouse** [comment]
Ok so that makes more sense.  The solar sensor is reading -100C in your output but the water temps were fine.  That threw me.

Pull njsPC.
`cd nodejs-poolController`
`git pull`
`npm i`

Then shutdown njsPC and restart it.

In the next day or so I will put together some instructions for the replay I need.

**Comment by rstrouse** [comment]
No.  If the electrical system is the same as it is here in the US, neutral is supplied from the main panel.   The IntelliCenter load center should have a neutral bus bar if the electrical system is the same.  It most certainly is not ground.

**Comment by rstrouse** [comment]
For US based systems the neutral bus is pointed to by the arrow if the electrician pulled a neutral from the main service panel.  Most commonly this is a white wire.
![image](https://user-images.githubusercontent.com/47839015/137171368-61a6314e-dc46-4ffd-ab42-4d6bf543ce88.png)

**Comment by mzuniga51** [comment]
Yes.  Both sensors are working correctly.  This is what I'm seeing:
<img width="563" alt="Screen Shot 2021-10-13 at 15 25 43" src="https://user-images.githubusercontent.com/91519307/137214871-43461bcd-dab2-476d-8bc9-884d948e7c82.png">

<img width="793" alt="Screen Shot 2021-10-13 at 15 26 03" src="https://user-images.githubusercontent.com/91519307/137214878-cf2a6ede-894e-438b-bad6-e62f6be3eac4.png">

**Comment by rstrouse** [comment]
Wait the pool circuit is not showing as on.

**Comment by mzuniga51** [comment]
Never mind.  I realized that if Pool at the OCP is off i can turn it on from njsPC and it will appear as on there.  If Pool is on at the OCP, then Pool will appear as off at njsPC and I will not be able to control it.  So, I have control of Pool from njsPC only when Pool has initially been turned on at njsPC

**Comment by rstrouse** [comment]
Are you using the IntelliCenter web app to turn the pool on and off or are you setting the OCP to service mode then turning it on and off?

**Comment by mzuniga51** [comment]
So after the pull, start with everything off and then turn on with app or
OCP again right?

**Comment by mzuniga51** [comment]
Here
[Uploading replay 3.zip…]()

**Comment by rstrouse** [comment]
Your upload of the replay failed.  The file upload didn't finish before you closed browser.

**Comment by mzuniga51** [comment]
Yes.  Service Mode is displayed
[replay 5.zip](https://github.com/tagyoureit/nodejs-poolController/files/7355944/replay.5.zip)
![4FDF316B-2259-44B1-8360-104C52828A71](https://user-images.githubusercontent.com/91519307/137551146-4075da93-becd-43e6-9e7c-a2f660e9b7ec.jpeg)

**Comment by rstrouse** [comment]
I know it is displayed there.  Is it displayed in dashPanel?

**Comment by mzuniga51** [comment]
Ohh, I see what you are getting at.  It does not appear in dashPanel!

**Comment by rstrouse** [comment]
Manuel, hi I am Robert from Northern California.  Please pull njsPC and lets see if the byte setting changes I made make a difference with an i10D.

**Comment by mzuniga51** [comment]
Hey Robert.  Did the last capture give you some additional insights?  Thanks

**Comment by rstrouse** [comment]
I have been a bit busy but I did take some time to look at the replay.   Here is what I am seeing happening.

Given the message sequence below.
![image](https://user-images.githubusercontent.com/47839015/140977444-5c04bbdd-77cf-4af8-b1f2-20798b375057.png)

1. The line with the black arrow and the poo icon is where you pressed the button to turn on the pool circuit.
2. Message number 35234 is njsPC requesting the circuit to be turned on.  The 32 on the 4th position of the payload indicates this.
3. Message number 35235 is the response from the OCP indicating that it received the message and is processing it.
4. Message number 35236 is njsPC asking the OCP which circuits are currently on.
5. Message number 35237 is the OCP responding with the current circuit status.  Notice the 32 on the 3rd position.
6. Message number 35238 is njsPC responding that it successfully got the message
7. Message number 35239 is the OCP turning the circuit back off again.  Notice that message 35239 and 35234 are the same command except the byte at the 4th position is 0 not 32.   At the end of this sequence is also a pattern 0,1,1,0 instead of the 0,0,1,0 which that byte that is 1 typically indicates that the circuit change is due to a schedule.

If when it does this can you hit the button again in dashPanel to get the pool circuit to stay on?

**Comment by mzuniga51** [comment]
Hi Robert.  Hmmm, can you rephrase the question?  Im not sure i’m
following.  Thanks

**Comment by rstrouse** [comment]
Using dashPanel I would like you to turn the pool circuit on.  When the OCP turns it back off again, hit the button again waiting 3 seconds between each attempt.  Do this at least 5 times.  I want to know if your OCP is persistent at refusing to turn on the pool.  It is so odd that it is only the pool circuit that will not remain on even after the OCP responds that it is turning it on.

**Comment by mzuniga51** [comment]
Robert, remember that when Pool is off in the OCP I have no problem turning on and off dashPanel.  See [replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/7523750/replay.zip)

The problem is when Pool is on because a schedule has activated it.  Then the on button on dashPanel appears gray while Pool is actually on. Then I have no control from dashPanel, presumably because dashPanel thinks Pool is off when it is actually on.  So OCP and dashPanel seem to fall out of sync for some reason.  See [replay-2.zip](https://github.com/tagyoureit/nodejs-poolController/files/7523749/replay-2.zip)

**Comment by rstrouse** [comment]
So replay.zip works fine and replay-2.zip does not?

**Comment by mzuniga51** [comment]
Correct

**Comment by rstrouse** [comment]
Hmmm....  there is something different here because the initial setup was not setting the bits correctly on the status message.  Now it appears that all this information is correct and it operates just like every other IntelliCenter model that I have seen.  Did you change anything in the configuration since we started looking into this?

**Comment by mzuniga51** [comment]
No.  Nothing Robert.  Sometimes it seems as there are almost random things
happening with this system.  Hope Pentair get his s@*t together with an
upcoming (?) firmware update.

**Comment by rstrouse** [comment]
Are you operating the OCP by switching to Service Mode?  If you are I think I may have been chasing the wrong thing.  Service mode is pretty much a legacy holdout from the older Pentair panels.  While it still has the benefit of disabling external controllers, it is not required for normal operation on an IntelliCenter panel.  In IntelliTouch you could not operate the relays at the OCP without service mode but that it no longer the case with IntelliCenter since it contains a localized control panel.

**Comment by mzuniga51** [comment]
No.  I stopped using service mode when you instructed me a while back.
But schedule activation in OCP (without user intervention) seem to have the
same effect as coming out of service mode to auto.

**Comment by rstrouse** [comment]
Do you have Manual OP Priority turned on?

**Comment by mzuniga51** [comment]
No I dont.  Should I?

**Comment by rstrouse** [comment]
No it doesn't work like they have it documented so I don't know if they have a bug with this or not.  Bear in mind that the other OCPs out there are also not bug free.  The main difference with IntelliCenter is that you can update the firmware without sending the panel off to have it updated.   I was a long time IntelliTouch user and lived with some vexing issues that would cause me to have to completely reprogram the system over and over again.  

I changed everything back so that i10D looks at the status messages just like every other IntelliCenter panel.  Given the last two sets of replays this is how IntelliCenter operates for every other panel we have seen.  Including another i10D.

So pull njsPC and dashPanel then restart both server applications.  After you have done that lets make sure the state is reflecting the state when you use the IntelliCenter web.  Then lets try it by using the control panel on the OCP (not service mode) making sure it shows the proper state in dashPanel.  Then the final thing will be to try to turn it off after it kicks on during a schedule then back on again.

**Comment by mzuniga51** [comment]
So, Robert, I was about to pull njsPC and try to access dashPanel only to find this.  Nothing to do with the problem we were dealing with?  I havent changed a thing.  Any idea what could have happened?  

Things I’ve tried:  raspberry reboot, intellicenter reboot, verify physical conection cable continuity, all lights in adapter working,  intellicenter factory reset,  Raspberry factory reset and package reinstall.  Nothing works!!

I
![4066072A-6098-4947-9F33-C10661E10A9F](https://user-images.githubusercontent.com/91519307/141660850-6e30c1c1-6f42-4dbf-9bc3-fe74aa043c26.jpeg)
![84107D02-BAC4-4487-834C-0F1755A8B77B](https://user-images.githubusercontent.com/91519307/141660854-4ab320d4-316d-4368-9a4b-dbc9cf6bc483.jpeg)

**Comment by mzuniga51** [comment]
Only additional things that occcur to me is a busted usb adapter or some corruption to the Intellicenter firmware 🤦🏻‍♂️🤷🏻‍♂️

**Comment by rstrouse** [comment]
The things that can cause this are in the following order.  Some of these you have already verified to not be the case.
1. Verify that there is nothing else accessing the adapter.  This includes more than one running instance of njsPC or terminal programs.
2. Wiring between the adapter and the comm port.
3. The transceiver on the RS485 adapter is set to TTY and not RS485.

I have some questions for you
Which adapter do you have?  
How are you launching njsPC?
Are you running any terminal programs?
Do you have bluetooth turned on on the Pi?

You should open dashPanel and navigate to the RS485 tab.  If the adapter is receiving successfully you should see the Received bytes incrementing as well as the successful numbers increasing.
![image](https://user-images.githubusercontent.com/47839015/141661980-a6fe8dff-cc7d-4107-bdd7-e059580b5c55.png)

**Comment by mzuniga51** [comment]
So, in the same order.

1.  Only one version running
2. Seem to be ok.  Black to ground, green to B-, yellow to A+.  Same cable and connected to the same COM port as before
3. Where do you check on this? 

This is my adapter https://www.amazon.com/gp/product/B081MB6PN2/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1
I'm currently launching manually (npm start)
Only things running is dashPanel and poolController.  VNC is also activated
Bluetooth is turned on

Adapter is receiving well with 0% failure rate

**Comment by mzuniga51** [comment]
Please feel free to ingnore if I'm imposing too much, but do you want to access my Raspberry directly?  I'm using VNC and my credentials are as follows:

login:  zuniga.manuel@gmail.com
password:  corderitos

Password for Pi is:  corderitos

Feel free to roam around if you think is useful

**Comment by mzuniga51** [comment]
Hey, please ignore if it is too much of an imposition but you can enter my Pi directly and roam around if you want.   Please feel free to make any changes or tests you want.  I'm sending my credentials in a pm in TFP.

**Comment by mzuniga51** [comment]
Robert, I see you tried to get in this morning but we had a blackout here.  Everything is on if you want to give it a try.  Many thanks!

**Comment by mzuniga51** [comment]
Robert, see you tried to get in and VNC blocked you based on your location.  I have authorized you now. Sorry about that.

**Comment by rstrouse** [comment]
So I looked at your pi and njsPC is opening the serial port successfully.  However, there is no traffic on it whatsoever.  The only bytes received are the ones that are part of the open for the port that are read from the buffer on the RS485 chip which points to the adapter not receiving information over the wire or the adapter is bad.  However, before we rule the adapter dead there are some other things to try.

I have disabled the remote GPIO and serial console on the pi.  These sometimes interfere with serial devices depending on the OS build.  I also tried to pull from the repository.  However, I am not sure how you installed the repository.  Did you do a git clone to install njsPC or did you load the software in another way?
`git clone https://github.com/tagyoureit/nodejs-poolController.git`

**Comment by mzuniga51** [comment]
First, thanks a lot.  You are the best.  My buddy reinstall my pi and the
packages for me so I´ll as him how he do it and will get back to you.
What adapter you use? I'm thinking to buy one now and have it as backup
even if we somehow manage to get the current one working.

**Comment by rstrouse** [comment]
I have quite a few but I do not recommend the standalone adapter that I currently use in my production as there have been tons of report that the quality has waned on them and even getting one that works initially is questionable. [JBtek](https://www.amazon.com/JBtek-Converter-Adapter-ch340T-Supported/dp/B00NKAJGZM/ref=sr_1_1?ie=UTF8&qid=1524495023&sr=8-1&keywords=jbtek+rs485)  When I get some time I am going to switch over to the Sequent MEGA-IND board that is already installed on my system.  I use that board as a power supply and input device and finally got around to creating the support functions to enable the RS485 port on that board.  

Honestly, it almost looks like the yellow and green wires are reversed on your adapter because any bytes that are getting loaded into the buffer appear to be scrambled, but you said it was working before.  If there is a chance that the wires got reversed reverse them at the adapter and see if that changes anything.  It won't hurt because yellow/green do not have a polarity by themselves.

Also here is an install guide that is a bit more descriptive.  From what I see of your install all you need are the prerequisites, njsPC, and dashPanel
https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#install-software-prerequisites

**Comment by mzuniga51** [comment]
So, good news is that I have managed to reconnect by changing the adapter.
Apparently the old one was busted.

No, not so good news is that things are a bit worse with dashPanel.  I'll
make some more comprehensive tests tomorrow and get back to you.

**Comment by mzuniga51** [comment]
Robert, I have prepared a [video](https://share.icloud.com/photos/0sk3b1HzDja9D0L8mktWWgHtg) that might be helpful explaining what I'm seen now.  There is a [capture](https://github.com/tagyoureit/nodejs-poolController/files/7549229/replay.zip) I made during the filming on the video so you can trace what I was doing step by step.  As you see, I think we backtracked a bit.  Again, manual priority is off and I'm not touching the service button at all.  Thanks

**Comment by rstrouse** [comment]
Your video never finished uploading.  However, I made some changes so lets see where we are after you pull njsPC.

**Comment by mzuniga51** [comment]
Sorry abt the video.  Anyway with the latest changes things are back at where we were before the adapter meltdown so, better than this afternoon.  I'd say the only issue is still schedules.  This capture shows the exact moment a schedule starts (pool started Off) .   At this precise moment Pool gets and stays green in OCP, web and app.   On the other hand, dashPanel gets momentarily green and then gray in dashPanel.   I don't get any response from that moment on by pressing the Pool button at the dashPanel.
[replay (3).zip](https://github.com/tagyoureit/nodejs-poolController/files/7551292/replay.3.zip)

**Comment by rstrouse** [comment]
Ok so here is the craziness with this.  The panel does not report that it it turned anything on when the schedule kicks in.  

Strangely, it does not report on the status message or the heartbeat like all other OCPs we have witnessed and the only circuit which it ignores is circuit 6 (your pool circuit).  It does however report on another byte when the pool circuit is manually turned on and occasionally on the status message (where all other OCPs report their relay status).   Interestingly, it always reports correctly when njsPC asks for the state information.  This is why it momentarily sets the correct state only to have the OCP incorrectly report.  It could be that the Pentair controllers do not see this because they simply ignore the state for body circuits.

So what I have done is completely ignore the heartbeat and status messages when it comes to the pool circuit and only rely on the requests made by njsPC or requests by other controllers to change the state.  Please pull njsPC and lets see if we slayed this dragon.

**Comment by mzuniga51** [comment]
This seem to be a hydra rather than a dragon.  One head cut and another
pops out.  Believe it or not, now Pool works fine BUT, wait for it, Spa is
doing now exactly what Pool was doing before.  Can you believe it?  I will
have to send you a very nice bottle of scotch if you solve this one.

**Comment by mzuniga51** [comment]
Oh, and in a bit of good news, today my second Intellichlor for my spa was installed. I can correctly read chemistry from both Intellichlors in dashPanel!!
<img width="824" alt="Screen Shot 2021-11-17 at 21 43 41" src="https://user-images.githubusercontent.com/91519307/142348103-0d466a86-b7c2-4afa-8cd6-8e6a093429b9.png">

**Comment by rstrouse** [comment]
That is crazy.  So far in all of this we have had the OCP reporting the pool circuit correctly on the status message.  Then it reported on the heartbeat message.  And finally, it reported only when a schedule did not turn it on.  All this time all other circuits and features reported just like every other panel out there including the other body circuit 1(spa).  Now it appears the spa is acting like the pool circuit.

The good news is that I have a pretty good idea how to fix it in short order.  So pull njsPC and lets see if our spa circuit works as it should.

**Comment by mzuniga51** [comment]
Well Robert, dragon duly slayed!   Everything seem to be finally in working
order.  I don't quite understand how you pull this off but I'm glad to have
had your help.

As a general note, It seem to me that Dual Body is not really ready for
prime time.  For example yesterday that I installed the second intellichem
suddenly the OCP got all confused and Pool circuit started to control Spa's
relay and Spa circuit got kind of stuck.  Very weird.  Resetting and power
off the whole thing did nothing so I had to factory reset the OCP and now
things are back to normal  (I'm quite expert at configuring the whole thing
from scratch now).  Particularly Schedules is something that seem to throw
the whole system a bit out of whack.  Hopefully new firmware from Pentair
would solve some of the issues.   Anyways, just venting I guess.

Two questions:  1) Is there an Amazon locker close where I can send a
little but well deserved token of appreciation? and 2) I have other minor
observations about dashPanel that I would like to pass to you as well as a
couple feature requests.  What is the best way to pass those to you?

I`ll keep monitoring the system and get back to you in case I have any
other issues.  I really appreciate all the effort and help so many many
thanks!

Manuel,

**Comment by rstrouse** [comment]
Yeah your panel is a bit of a mystery.  The reason why I say that is because the i5P is the base for all IntelliCenter systems.  All the software and programming logic exists on that board and every IntelliCenter system has one.  This means that an i10D is an i5P with some additional relays and sensors.  All the other personality cards including the i10D do not contain any programming logic.  The micro-controller, firmware, and base equipment control is the same on every system.  The way the base controller knows which personality card is attached is simply by checking a circuit that has a different R value for each card.

Given all of the peripherals on the RS485 bus are universal no matter which personality card is attached so the communication over RS485 must be consistent or they will not work.   So there are only 2 potential things that could be causing weirdness.

1.  Pentair doesn't honor its own status outputs and is acquiring the relay status from the other 2 message outputs.   This is likely given the fact that indoor and wireless control panels would not work properly on an i10D if they didn't.  It might also explain why the wireless controller loses the configuration sometimes when it sleeps and the wakeup process takes a bit of time to get its poop in a group.  It has to ask for all the information that it didn't have when it was in slumber.
2. There is a different hardware version of the i5P motherboard for your region.  I do find it odd that on my IntelliCenter, I cannot change change from Fahrenheit to Celsius.  Then again I have not messed with it at the OCP so perhaps it can only be done there.  Can you change from C to F on the Pentair web interface.
3. The i10D body control (circuit 1 & 6) functions have some unintended software features (bugs) that do not set the status bytes properly in i10D mode.  This is probably what is going on.   Depending on whether the personality card is Shared, Single, or Dual these two circuits are managed differently.  The i10D team may be the folks with 6 fingers on the left hand and 4 on the right.  Hopefully Pentair releases some fixes soon.

For the dashPanel stuff you can simply post enhancement requests or unintended software features on the repo [dashPanel Issues](https://github.com/rstrouse/nodejs-poolController-dashPanel/issues).  Thanks for the offer to send me something but I really don't need anything at this time.  I got involved in this project due to my own needs and as it turns out others seemed to have them as well.  Bonus.

**Comment by mzuniga51** [comment]
Surely either of the three or a combination of all of them.  I can change
from C to F both in the OCP, app and web, as well as from 12 to 24h time
format but solutions seem half baked.  So much so that this time around I
decided to stay in F and 12h.    Is there any idea as to when the next
firmware update will be released?

Thank you again for all your kind help and let's keep in touch.

Manuel

**Comment by rstrouse** [comment]
I have no idea when the next firmware is going to be released.   I also have no idea why the temperature units are not enabled for me.  It is truly strange.

It seems so weird that the temps on the spa are so far out in left field.  I know for a fact that your spa isn't 79C without a roast, carrots, and a few potatoes floating around in it.  Certainly there are no people in it at that temperature unless they are simmering.  It's almost as if half the temps are in C and half in F.

It could also just be a poor connection but you will have a heck of a time setting the spa temperature with sensor being so far off.

**Comment by mzuniga51** [comment]
Weird in deed.  Better work in F and just forget about it.

Hey, just a quick question.   Are there specific parameters for using
custom backgrounds?  I'm trying to upload a photo of my pool but the
"Uploading Background File" does nothing.

**Comment by rstrouse** [comment]
No there aren't any except that it needs to be a valid image file (png, jpg, bmp, gif ...etc).  When you click the Upload Image does the following dialog appear?

![image](https://user-images.githubusercontent.com/47839015/142697709-fe94786b-5974-4b69-8235-24edd780875f.png)

After that you need to click on the popup button at the end of the field to select a the file to upload.  If this doesn't upload you can copy the file onto your `nodejs-poolController-dashPanel/themes/images` directory and it will become available in the dropdown.

**Comment by mzuniga51** [comment]
Awsome.  I will go ahead and close this issue.  Have a great day.
